aptos-core/consensus/src/block_storage/sync_manager.rs

```diff
@@ -318,6 +319,67 @@ impl BlockStore {
         Ok(())
     }
 
+    pub async fn fast_forward_sync_by_epoch(
+        &self,
+        mut retriever: BlockRetriever,
+        epoch: u64,
+    ) -> anyhow::Result<()> {
+        let highest_commit_cert = self.highest_commit_cert();
+        let payload_manager = self.payload_manager.clone();
+        let storage = self.storage.clone();
+        let (blocks, mut quorum_certs, ledger_infos) = retriever
+            .retrieve_block_by_epoch(
+                epoch,
+                highest_commit_cert.commit_info().id(),
+                retriever.validator_addresses(),
+                payload_manager.clone(),
+            )
+            .await?;
+
+        for (i, block) in blocks.iter().enumerate() {
```

Comment:
Could you please add some metrics like BLOCKS_FETCHED_FROM_NETWORK_WHILE_FAST_FORWARD_SYNC ?

Reply:
This variable already exists, CUR_BLOCK_SYNC_BLOCK_SUM_GAUGE